app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala

```diff
@@ -0,0 +1,298 @@
+      case CgtRef(_) =>
+        new URL(
+          s"$ifBaseUrl/registration/relationship?idtype=ZCGT&ref-no=$encodedClientId&agent=false&active-only=true&regime=${getRegimeFor(taxIdentifier)}&relationship=ZA01&auth-profile=ALL00001")
+    }
+
+    getWithIFHeaders("GetActiveClientRelationships", url, ifAuthToken, ifEnv).map { response =>
+      response.status match {
+        case Status.OK =>
+          response.json.as[ActiveRelationshipResponse].relationship.find(isActive)
+        case Status.BAD_REQUEST                             => None
+        case Status.NOT_FOUND                               => None
+        case _ if response.body.contains("AGENT_SUSPENDED") => None
+        case other: Int =>
+          throw UpstreamErrorResponse(response.body, other, other)
+      }
+    }
+  }
+
+  // IF API #1168
+  def getInactiveClientRelationships(taxIdentifier: TaxIdentifier)(
+    implicit hc: HeaderCarrier,
+    ec: ExecutionContext): Future[Seq[InactiveRelationship]] = {
+    val encodedClientId = UriEncoding.encodePathSegment(taxIdentifier.value, "UTF-8")
+
+    val url = inactiveClientRelationshipIFUrl(taxIdentifier, encodedClientId)
+
+    getWithIFHeaders("GetInactiveClientRelationships", url, ifAuthToken, ifEnv).map { response =>
+      response.status match {
+        case Status.OK =>
+          response.json.as[InactiveRelationshipResponse].relationship.filter(isNotActive)
+        case Status.BAD_REQUEST => Seq.empty
+        case Status.NOT_FOUND   => Seq.empty
+        case other: Int =>
+          throw UpstreamErrorResponse(response.body, other, other)
+      }
+    }
+  }
+
+  // IF API #1168 (for agent)
+  def getInactiveRelationships(
+    arn: Arn)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[InactiveRelationship]] = {
+    val encodedAgentId = UriEncoding.encodePathSegment(arn.value, "UTF-8")
+    val now = LocalDate.now().toString
+    val from: String = LocalDate.now().minusDays(showInactiveRelationshipsDays).toString
+    val regime = "AGSV"
+    val url = new URL(
+      s"$ifBaseUrl/registration/relationship?arn=$encodedAgentId&agent=true&active-only=false&regime=$regime&from=$from&to=$now")
+
+    val cacheKey = s"${arn.value}-$now"
+    agentCacheProvider.agentTrackPageCache(cacheKey) {
+      getWithIFHeaders(s"GetInactiveRelationships", url, ifAuthToken, ifEnv).map { response =>
+        response.status match {
+          case Status.OK =>
+            response.json.as[InactiveRelationshipResponse].relationship.filter(isNotActive)
+          case Status.BAD_REQUEST                             => Seq.empty
+          case Status.NOT_FOUND                               => Seq.empty
+          case _ if response.body.contains("AGENT_SUSPENDED") => Seq.empty
+          case other: Int =>
+            throw UpstreamErrorResponse(response.body, other, other)
+        }
```

Comment:
and these

Reply:
using same pattern as DES but good that you spotted this

Reply:
:+1: